### ISSUE TITLE
ZDF show info api is broken

### ISSUE THREAD

**Issue by cemrich** [cemrich/zapp#41]
Due to the recent relaunch the api stopped returning any shows. The new api (https://api.zdf.de/cmdm/epg/broadcasts) seems to be protected.

**Comment by cemrich** [comment]
Okay, the API token seems to be static and can be obtained from https://www.zdf.de/ZDFplayer/configs/zdf/zdf2016/configuration.json

A basic Request looks like this:

    curl "https://api.zdf.de/cmdm/epg/broadcasts?from=2016-11-01T20%3A58%3A42Z&limit=1&page=1&tvServices=zdf&order=desc" -H "Host: api.zdf.de" -H "Accept: application/vnd.de.zdf.v1.0+json" -H "Api-Auth: Bearer d2726b6c8c655e42b68b0db26131b15b22bd1a32" -H "Origin: https://www.zdf.de"

However, I get responses that are not reproducible with random channels and dates. If anyone could figure out how this api is supposed to work I would appreciate the feedback. For the time being I will deactivate the old ZDF downloader

**Comment by cemrich** [comment]
Fallbacks for some channels could be:

- [ ] https://www.phoenix.de/php/mediaplayer/data/get_livestream.php?ch=1&system=android for phoenix

**Comment by juergenhoetzel** [comment]
> curl "https://api.zdf.de/cmdm/epg/broadcasts?from=2016-11-01T20%3A58%3A42Z&limit=1&page=1&tvServices=zdf&order=desc" -H "Host: api.zdf.de" -H "Accept: application/vnd.de.zdf.v1.0+json" -H "Api-Auth: Bearer d2726b6c8c655e42b68b0db26131b15b22bd1a32" -H "Origin: https://www.zdf.de"
> However, I get responses that are not reproducible with random channels and dates. If anyone could figure out how this api is supposed to work I would appreciate the feedback. For the time being I will deactivate the old ZDF downloader

I investigated the web player and found out, you need to supply the `to` query parameter instead of `from`. This works for me:

```bash
curl "https://api.zdf.de/cmdm/epg/broadcasts?to=2017-03-11T12%3A58%3A42Z&limit=1&page=1&tvServices=zdf&order=desc" -H "Host: api.zdf.de" -H "Accept: application/vnd.de.zdf.v1.0+json" -H "Api-Auth: Bearer d2726b6c8c655e42b68b0db26131b15b22bd1a32" -H "Origin: https://www.zdf.de"|json_pp
```

```json
{
   "http://zdf.de/rels/cmdm/broadcasts" : [
      {
         "duration" : 15480,
         "withChat" : false,
         "geolocationVOD" : null,
         "text" : "<b>Eisschnelllauf-Weltcup, Finale</b><br/>500 m Herren<br/>Übertragung aus<br/>Stavanger/Norwegen<br/>Reporter: Hermann Valkyser<br/><br/>ca. 11:05 Uhr<br/><b>Weltcup Nordische Kombination</b><br/>Springen<br/>Übertragung aus<br/>Oslo/Norwegen<br/>Reporter: Volker Grube<br/>Moderation: Sven Voss<br/><br/>ca. 11:50 Uhr<br/><b>Biathlon-Weltcup</b><br/>12,5 km Verfolgung Herren<br/>Übertragung aus<br/>Kontiolahti/Finnland<br/>Reporter: Christoph Hamm, Herbert Fritzenwenger<br/>Moderation: Alexander Ruda<br/>Experte: Sven Fischer<br/><br/>ca. 12:57 Uhr<br/><b>Eisschnelllauf-Weltcup, Finale</b><br/>1000 m Herren, 3000 m Damen<br/>Zusammenfassung aus<br/>Stavanger/Norwegen<br/>Reporter: Hermann Valkyser<br/><br/>ca. 13:45 Uhr<br/><b>Alpiner Ski-Weltcup</b><br/>Riesenslalom Damen<br/>Zusammenfassung aus<br/>Squaw Valley/USA<br/>Reporter: Aris Donzelli<br/><br/>ca. 14:10 Uhr<br/><b>Snowboard-WM</b><br/>Slopestyle<br/>Zusammenfassung aus<br/>Sierra Nevada/Spanien<br/>Reporter: Marc Windgassen<br/><br/>ca. 14:30 Uhr<br/><b>Langlauf-Weltcup</b><br/>50 km Herren<br/>Übertragung aus<br/>Oslo/Norwegen<br/>Reporter: Peter Leissl<br/><br/>ca. 15:02 Uhr<br/><b>Biathlon-Weltcup</b><br/>10 km Verfolgung Damen<br/>Übertragung aus<br/>Kontiolahti/Finnland<br/>Reporter: Christoph Hamm, Herbert Fritzenwenger<br/>Moderation: Alexander Ruda<br/>Experte: Sven Fischer<br/><br/>ca. 16:05 Uhr<br/><b>Weltcup-Skispringen</b><br/>Herren Team, 1. Durchgang<br/>Übertragung aus<br/>Oslo/Norwegen<br/>Reporter: Stefan Bier<br/>Moderation: Sven Voss<br/><br/>ca. 17:15 Uhr<br/><b>Weltcup-Skispringen</b><br/>Herren Team, 2. Durchgang<br/>Übertragung aus<br/>Oslo/Norwegen<br/>Reporter: Stefan Bier<br/>Moderation: Sven Voss<br/><br/>ca. 18:05 Uhr<br/><b>Weltcup Nordische Kombination</b><br/>10 km Langlauf<br/>Zusammenfassung aus<br/>Oslo/Norwegen<br/>Reporter: Volker Grube<br/>Moderation: Sven Voss<br/><br/>**Snowboard-WM bei zdfsport.de:<br/>Slopestyle aus Sierra Nevada/Spanien ab 12.00 Uhr im Livestream <br/><br/><br/>**Weltcup Nordische Kombination bei zdfsport.de:<br/>10 km Langlauf aus Oslo/Norwegen ab 15.40 Uhr im Livestream <br/><br/>**Alpiner Ski-Weltcup bei zdfsport.de:<br/>Slalom Damen aus Squaw Valley/USA im Livestream <br/>1. Lauf ab 19.00 Uhr <br/>2. Lauf ab 22.00 Uhr<br/><br/>**Snowboard-WM bei zdfsport.de:<br/>Halfpipe aus Sierra Nevada/Spanien ab 20.00 Uhr im Livestream",
         "playoutId" : null,
         "stereo" : true,
         "modified" : "2017-03-11T13:05:38+01:00",
         "profile" : "http://zdf.de/rels/cmdm/broadcast",
         "effectiveAirtimeEnd" : null,
         "http://zdf.de/rels/tvservice" : {
            "profile" : "http://zdf.de/rels/content/conf-tv-service",
            "tvServiceLogoRef" : {
               "title" : "Senderlogo ZDF",
               "profile" : "http://zdf.de/rels/image",
               "layouts" : {
                  "840x280" : "https://www.zdf.de/assets/2400_ZDF-100~840x280?cb=1479296030779",
                  "1500x800" : "https://www.zdf.de/assets/2400_ZDF-100~1500x800?cb=1479296030779",
                  "240x270" : "https://www.zdf.de/assets/2400_ZDF-100~240x270?cb=1479296030779",
                  "314x314" : "https://www.zdf.de/assets/2400_ZDF-100~314x314?cb=1479296030779",
                  "1900x400" : "https://www.zdf.de/assets/2400_ZDF-100~1900x400?cb=1479296030779",
                  "640x720" : "https://www.zdf.de/assets/2400_ZDF-100~640x720?cb=1479296030779",
                  "380x170" : "https://www.zdf.de/assets/2400_ZDF-100~380x170?cb=1479296030779",
                  "1920x1080" : "https://www.zdf.de/assets/2400_ZDF-100~1920x1080?cb=1479296030779",
                  "840x360" : "https://www.zdf.de/assets/2400_ZDF-100~840x360?cb=1479296030779",
                  "1900x200" : "https://www.zdf.de/assets/2400_ZDF-100~1900x200?cb=1479296030779",
                  "2400x1350" : "https://www.zdf.de/assets/2400_ZDF-100~2400x1350?cb=1479296030779",
                  "1280xauto" : "https://www.zdf.de/assets/2400_ZDF-100~1280xauto?cb=1479296030779",
                  "276x155" : "https://www.zdf.de/assets/2400_ZDF-100~276x155?cb=1479296030779",
                  "384xauto" : "https://www.zdf.de/assets/2400_ZDF-100~384xauto?cb=1479296030779",
                  "1500x600" : "https://www.zdf.de/assets/2400_ZDF-100~1500x600?cb=1479296030779",
                  "384x216" : "https://www.zdf.de/assets/2400_ZDF-100~384x216?cb=1479296030779",
                  "760x340" : "https://www.zdf.de/assets/2400_ZDF-100~760x340?cb=1479296030779",
                  "1152x1296" : "https://www.zdf.de/assets/2400_ZDF-100~1152x1296?cb=1479296030779",
                  "768x432" : "https://www.zdf.de/assets/2400_ZDF-100~768x432?cb=1479296030779",
                  "1280x720" : "https://www.zdf.de/assets/2400_ZDF-100~1280x720?cb=1479296030779",
                  "1900x570" : "https://www.zdf.de/assets/2400_ZDF-100~1900x570?cb=1479296030779",
                  "1500x300" : "https://www.zdf.de/assets/2400_ZDF-100~1500x300?cb=1479296030779",
                  "original" : "https://www.zdf.de/assets/2400_ZDF-100~original?cb=1479296030779",
                  "840x140" : "https://www.zdf.de/assets/2400_ZDF-100~840x140?cb=1479296030779",
                  "768xauto" : "https://www.zdf.de/assets/2400_ZDF-100~768xauto?cb=1479296030779"
               },
               "source" : "ZDF",
               "self" : "/content/documents/zdf/bilder/2400_ZDF-100.json",
               "canonical" : "/content/documents/2400_ZDF-100.json",
               "altText" : "Senderlogo ZDF"
            },
            "tvServiceId" : "ZDF",
            "tvServiceTitle" : "ZDF",
            "canonical" : "/content/documents/zdf-114.json",
            "self" : "/content/documents/zdf/sender/zdf/zdf-114.json"
         },
         "newProgramData" : false,
         "visibleFrom" : "2017-02-18T12:57:00+01:00",
         "http://zdf.de/rels/cmdm/programme-item" : "/cmdm/epg/programme-items/POS_bffb406c-40fb-4ae0-adfa-97e2b9fc329b",
         "title" : "ZDF SPORTextra - Wintersport",
         "dolbyDigital51" : false,
         "airtimeEnd" : "2017-03-11T15:00:00+01:00",
         "partId" : 2,
         "live" : false,
         "effectiveAirtimeBegin" : null,
         "pharosId" : "D1EF63CCB3536B04",
         "livestream" : false,
         "primaryBrand" : "ZDF SPORTextra",
         "mono" : false,
         "primaryBrandId" : "145f6a3a-10b7-3a31-99dd-c3396369b148",
         "audioComments" : false,
         "subtitle" : null,
         "airtimeBegin" : "2017-03-11T12:57:00+01:00",
         "youtubeRight" : false,
         "foreignLangWithCaption" : false,
         "widescreen16_9" : false,
         "http://zdf.de/rels/image" : {
            "altText" : "Logo \"ZDF SPORTextra Winter\".",
            "layouts" : {
               "276x155" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=276x155",
               "1920x1080" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=1920x1080",
               "2400x1350" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=2400x1350",
               "1280x720" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=1280x720",
               "640x720" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=640x720",
               "1152x1296" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=1152x1296",
               "240x270" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=240x270",
               "768x432" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=768x432",
               "384x216" : "https://epg-image.zdf.de/fotobase-webdelivery/images/2c76f388-d1dc-4308-ad7e-51a723fc6eb5?layout=384x216"
            },
            "source" : "Corporate Design",
            "profile" : "http://zdf.de/rels/image"
         },
         "newAirtime" : false,
         "tvServiceId" : "1",
         "visibleTo" : "2018-03-11T12:57:00+01:00",
         "onlineTo" : null,
         "fsk" : null,
         "http://zdf.de/rels/cmdm/broadcasts-parts" : [
            {
               "partId" : 1,
               "airtimeDate" : "2017-03-11+01:00",
               "self" : "/cmdm/epg/broadcasts/bffb406c-40fb-4ae0-adfa-97e2b9fc329b?profile=part",
               "airtimeBegin" : "2017-03-11T10:42:00+01:00",
               "profile" : "http://zdf.de/rels/cmdm/broadcast-part",
               "airtimeEnd" : "2017-03-11T12:55:00+01:00"
            },
            {
               "partId" : 2,
               "airtimeDate" : "2017-03-11+01:00",
               "airtimeBegin" : "2017-03-11T12:57:00+01:00",
               "self" : "/cmdm/epg/broadcasts/05822616-bcf6-4541-b153-208437f0ff31?profile=part",
               "profile" : "http://zdf.de/rels/cmdm/broadcast-part",
               "airtimeEnd" : "2017-03-11T15:00:00+01:00"
            },
            {
               "airtimeDate" : "2017-03-11+01:00",
               "airtimeBegin" : "2017-03-11T15:02:00+01:00",
               "self" : "/cmdm/epg/broadcasts/fd6cacc7-14e7-4631-9eed-d79a26a83ac4?profile=part",
               "partId" : 3,
               "airtimeEnd" : "2017-03-11T17:10:00+01:00",
               "profile" : "http://zdf.de/rels/cmdm/broadcast-part"
            },
            {
               "airtimeEnd" : "2017-03-11T18:35:00+01:00",
               "profile" : "http://zdf.de/rels/cmdm/broadcast-part",
               "self" : "/cmdm/epg/broadcasts/8b3d45ea-8fff-4a2c-91f2-d947f08b7f4f?profile=part",
               "airtimeBegin" : "2017-03-11T17:15:00+01:00",
               "airtimeDate" : "2017-03-11+01:00",
               "partId" : 4
            }
         ],
         "tvService" : "ZDF",
         "geolocationLivestream" : null,
         "vpsBegin" : "2017-03-11T10:42:00+01:00",
         "onlineFrom" : null,
         "posId" : "05822616-bcf6-4541-b153-208437f0ff31",
         "blackwhite" : false,
         "signLanguage" : false,
         "self" : "/cmdm/epg/broadcasts/05822616-bcf6-4541-b153-208437f0ff31",
         "onlineFirst" : false,
         "airtimeDate" : "2017-03-11+01:00",
         "dualChannel" : false,
         "hd" : true,
         "dolbySurround" : false,
         "caption" : true,
         "subheadline" : null
      }
   ],
   "self" : "/cmdm/epg/broadcasts?to=2017-03-11T12%3A58%3A42Z&limit=1&page=1&tvServices=zdf&order=desc",
   "profile" : "http://zdf.de/rels/cmdm/resultpage-broadcasts",
   "next-archive" : "/cmdm/epg/broadcasts?to=2017-03-11T12%3A58%3A42Z&limit=1&page=2&tvServices=zdf&order=desc"
}
```

**Comment by cemrich** [comment]
Wow, this seems to work as expected! Thank you very much - I'll re-implement this as soon as I can.

**Comment by cemrich** [comment]
This issue will be fixed in v1.3.1. It will take a few days until the new version finds its way into fDroid.
